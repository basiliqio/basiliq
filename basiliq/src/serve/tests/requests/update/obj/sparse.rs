use super::*;

lazy_static::lazy_static! {
    pub static ref UPDATE_BODY: serde_json::Value =
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": 100
            })
        })
    });
}

crate::run_test_request!(
    no_fields,
    Method::PATCH,
    format!(
        "/public__peoples/{}?fields[public__peoples]=",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    one_fields,
    Method::PATCH,
    format!(
        "/public__peoples/{}?fields[public__peoples]=first-name",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    multi_fields,
    Method::PATCH,
    format!(
        "/public__peoples/{}?fields[public__peoples]=first-name,gender",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    unknown_fields,
    Method::PATCH,
    format!(
        "/public__peoples/{}?fields[public__peoples]=AAAAAAAAAAAAA",
        PEOPLES_IDS[0]
    ),
    400,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    unknown_type,
    Method::PATCH,
    format!(
        "/public__peoples/{}?fields[AAAAAAAAAAAAAA]=gender",
        PEOPLES_IDS[0]
    ),
    400,
    UPDATE_BODY.clone()
);
