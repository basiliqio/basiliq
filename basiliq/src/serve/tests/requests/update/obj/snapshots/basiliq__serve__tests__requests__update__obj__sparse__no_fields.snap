---
source: basiliq/src/serve/tests/requests/update/obj/sparse.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": {
    "type": "public__peoples",
    "id": "[uuid]"
  }
}
